password_security/controllers/main.py

```diff
@@ -1,42 +1,30 @@
 # Copyright 2015 LasLabs Inc.
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
 
-import operator
+import logging
+
+from werkzeug.exceptions import BadRequest
 
 from odoo import http
-from odoo.exceptions import UserError
 from odoo.http import request
 
 from odoo.addons.auth_signup.controllers.main import AuthSignupHome
-from odoo.addons.web.controllers.main import Session, ensure_db
-
+from odoo.addons.web.controllers.home import ensure_db
 
-class PasswordSecuritySession(Session):
-    @http.route()
-    def change_password(self, fields):
-        new_password = operator.itemgetter("new_password")(
-            dict(list(map(operator.itemgetter("name", "value"), fields)))
-        )
-        user_id = request.env.user
-        user_id._check_password(new_password)
-        return super(PasswordSecuritySession, self).change_password(fields)
+_logger = logging.getLogger(__name__)
 
 
 class PasswordSecurityHome(AuthSignupHome):
     def do_signup(self, qcontext):
         password = qcontext.get("password")
-        user_id = request.env.user
-        user_id._check_password(password)
-        return super(PasswordSecurityHome, self).do_signup(qcontext)
-
-    @http.route("/password_security/estimate", auth="none", type="json")
-    def estimate(self, password):
-        return request.env["res.users"].get_estimation(password)
+        user = request.env.user
+        user._check_password(password)
+        return super().do_signup(qcontext)
 
     @http.route()
     def web_login(self, *args, **kw):
         ensure_db()
-        response = super(PasswordSecurityHome, self).web_login(*args, **kw)
+        response = super().web_login(*args, **kw)
```

Comment:
On a big change like this, I would leave style items like this alone to make the diff smaller. Odoo codebase seems to prefer the explicit form anyway.